neurodsp/spectral/power.py

```diff
@@ -241,3 +244,47 @@ def compute_spectrum_medfilt(sig, fs, filt_len=1., f_range=None):
         freqs, spectrum = trim_spectrum(freqs, spectrum, f_range)
 
     return freqs, spectrum
+
+
+def compute_spectrum_multitaper(sig, fs, bandwidth, num_windows):
+    """Compute the power spectral density using the multi-taper method.
+
+    Parameters
+    ----------
+    sig : 1d or 2d array
+        Time series.
+    fs : float
+        Sampling rate, in Hz.
+    bandwith : float
+        Frequency bandwith of multi-taper window function.
+    num_windows : int.
+        Number of slepian windows used to weight the signal.
+
+    Returns
+    -------
+    freqs : 1d array
+        Frequencies at which the measure was calculated.
+    spectrum : 1d or 2d array
+        Power spectral density using multi-taper method.
+    """
+
+    from scipy.signal.windows import dpss
+
+    # Compute signal length based on input shape
+    sig_len = sig.shape[sig.ndim - 1]
+
+    # Create slepian sequences
+    slepian_sequences = dpss(sig_len, bandwidth/2, num_windows)
+
+    # Compute fourier on signal weighted by each slepian sequence
+    freqs = np.fft.rfftfreq(sig_len, 1. /fs)
+    spectrums = np.fft.rfft(slepian_sequences[:, np.newaxis]*sig)
```

Comment:
Does this need abs()**2 to move to psd? I think rfft gives fourier coefficients - some were negative and plotting log power return nans.

Reply:
I believe this is correct. After making this change the resulting spectra looks much similar to that of mne's `psd_array_multitaper` (shown below). Default parameters were used for `psd_array_multitaper`, whereas `bandwith = 8 * (sfreq / n_times)` and `num_windows = 7` in `compute_spectrum_multitaper`. I believe this is equivalent to the default values used in `psd_array_multitaper`.

![image](https://github.com/neurodsp-tools/neurodsp/assets/103983632/f5e3ea61-331d-4101-9586-d1e2d56831a1)


![image](https://github.com/neurodsp-tools/neurodsp/assets/103983632/dfd30b26-71ac-4be9-a8fc-dc77e21819a7)